### PR TITLE
Add v1.17.1

### PR DESCRIPTION
**Pull request by kenhys**
**Which issue(s) this PR fixes**: 

Fixes #

**What this PR does / why we need it**: 

Let's make a point release.

**Docs Changes**:

N/A

**Release Note**: 

N/A

### CODE REVIEW

**Comment by kenhys** [comment]
Not mentioned yet because of WIP

* #4580 
* #4586

**Comment by ashie** [comment]
>     * [Extend Embedded Ruby Code support for Hashes and Arrays #4580](https://github.com/fluent/fluentd/pull/4580)

It should be v1.18 or later as I mentioned in https://github.com/fluent/fluentd/pull/4580#issuecomment-2274824681

**Comment by kenhys** [comment]
Added last one piece for v1.17.1 in CHANGELOG.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,48 @@
...
+  https://github.com/fluent/fluentd/pull/4586
+
+### Bug Fix
+
+* logger: Fix LoadError with console gem v1.25
+  https://github.com/fluent/fluentd/pull/4492
+* parser_json: Fix wrong LoadError warning
+  https://github.com/fluent/fluentd/pull/4522
+* in_tail: Fix to skip lines with max line size limitation
+  https://github.com/fluent/fluentd/pull/4530
```
Comment (daipom) [inline CHANGELOG.md:31]: It is difficult to explain #4530 briefly..., but it should be more concrete for users to understand this fix.
How about this?

```suggestion
* in_tail: Fix an issue where a large single line could consume a large amount of memory even though `max_line_size` is set
  https://github.com/fluent/fluentd/pull/4530
```

**Review by daipom — COMMENTED** [review]
Thanks!
I have commented on some points.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,48 @@
...
+* out_http: Add `compress gzip` option
+  https://github.com/fluent/fluentd/pull/4528
+* in_exec: Add `encoding` option to handle non-ascii characters
+  https://github.com/fluent/fluentd/pull/4533
+* in_tail: Add throttling metrics
+  https://github.com/fluent/fluentd/pull/4578
+* compat: Improve method call performance
+  https://github.com/fluent/fluentd/pull/4588
+* Add logger gem dependency for Ruby 3.5
+  https://github.com/fluent/fluentd/pull/4589
```
[the remark this answers is not in the document]
Comment (kenhys) [reply CHANGELOG.md]: Thanks, fixed it.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,48 @@
...
+* in_sample: Add `reuse_record` parameter to reuse the sample data
+  https://github.com/fluent/fluentd/pull/4586
```
[the remark this answers is not in the document]
Comment (kenhys) [reply CHANGELOG.md:20]: It looks nice!

**Review by daipom — APPROVED** [review]
Sorry for my typo! Thanks for fixing it!
LGTM.

**Comment by kenhys** [comment]
Thank you for reviewing!

**Pull request merged**